### ISSUE TITLE
[BUG] uxListRemove hit a data abort issue.

### ISSUE THREAD

**Issue by lipengit** [FreeRTOS/FreeRTOS-Kernel#60]
**Describe the bug**
We hit a data abort issue in our application, which creates one event group, and different tasks can set the same event. For example, task1 is blocked by event1, where both task2 and task3 can set event1. When task2 set event1, task1 will be moved from blocked list to ready list. Before task1 is blocked by event1 again, if task3 set event1, we hit the data abort issue. 

We worked around this issue by making some small changes in tasks.c and list.c, It basically checks if 'pxItemToRemove->pxContainer' is empty before removing it. Please find the diff attached. [diff.txt](https://github.com/FreeRTOS/FreeRTOS-Kernel/files/4626137/diff.txt)

The changes have zero impact on the existing kernel code. I was wondering if these changes can be merged into the master branch?

**Comment by RichardBarry** [comment]
We would need a much deeper dive into your particular usage scenario before we were able to accept this as a bug.  It is of course possible there is a bug here, but on my first reading my initial thought is that there may be a problem in the way you are using the code (somehow breaking critical regions, using incorrect interrupt priorities, or something else) and your change is just masking the root cause rather than fixing it.  Please create a support request thread here https://forums.freertos.org/ where we will be happy to get to the bottom of the issue with you.  If it turns out to be a bug then we can either re-open this ticket or create a new one with a description of what the bug is and how to replicate it.  The description at the moment is a common use case for event groups, hence my thinking that we need to find out what is specific about your particular scenario.
